packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -365,29 +377,54 @@ export class AtomicCommerceInterface
   }
 
   private initSearchStatus() {
-    if (this.type === 'product-listing') {
-      this.searchStatus = buildProductListing(this.engine!);
-    } else {
-      this.searchStatus = buildSearch(this.engine!);
-    }
-    this.unsubscribeSearchStatus = this.searchStatus.subscribe(() => {
+    this.searchOrListing =
+      this.type === 'product-listing'
+        ? buildProductListing(this.engine!)
+        : buildSearch(this.engine!);
+
+    this.unsubscribeSearchStatus = this.searchOrListing.subscribe(() => {
       if (
-        !this.searchStatus.state.isLoading &&
+        !this.searchOrListing.state.isLoading &&
         this.store.hasLoadingFlag(FirstSearchExecutedFlag)
       ) {
         this.store.unsetLoadingFlag(FirstSearchExecutedFlag);
       }
     });
   }
 
+  private initSummary() {
+    this.summary =
+      this.type === 'product-listing'
```

Comment:
the type smell like "magic string", should we refactor it?

Reply:
`this.type` is strongly typed. 
You can't do `if this.type=='foo'`, TypeScript will complain.

So I don't think there's anything to refactor.